### PR TITLE
Add basic coverage support for Rust crates

### PR DESCRIPTION
**Pull request by TimDiekmann**
## 🌟 What is the purpose of this PR?

Adds coverage collecting for specified Rust projects

## 🔍 What does this change?

- Added a `cargo make` configuration to run coverage using `cargo-llvm-cov`
- Slightly changed the output of `setup.py` to use the directory name instead of the full path
- Added a pattern to find crates to collect coverage data for
- Added a CI configuration to run the configuration

## ⚠️ Known issues

- The Graph API currently only runs the unit tests and integration tests for coverage reporting.
- The coverage currently runs unconditionally. We need to enable the used flags as carried forward flags to enable conditional coverage reporting. This may be annoying to see Rust jobs in non-Rust PRs, but all coverage report jobs are faster than the backend integration test suite, which runs each time anyway.

### CODE REVIEW

File: .github/workflows/rust.yml [hunk 1]
```diff
@@ -179,6 +179,59 @@ jobs:
...
+    steps:
+      - name: Checkout source code
+        uses: actions/checkout@v3
+
+      - name: Install Rust
+        uses: actions-rs/toolchain@v1
+        with:
+          profile: minimal
+          toolchain: ${{ matrix.toolchain }}
+          override: true
```
Comment (kachkaev) [inline .github/workflows/rust.yml:199]: Can we use the same action everywhere? `action-rs/toolchain@v1` is [unmaintained](https://github.com/actions-rs/toolchain/issues/216). We could go for `./.github/actions/install-rust-toolchain` or some third-party alternative.

File: .github/workflows/rust.yml [hunk 2]
```diff
@@ -179,6 +179,59 @@ jobs:
...
+        shell: bash
+        run: |
+          cargo install cargo-quickinstall
+          cargo quickinstall cargo-make --version 0.35.15
+          cargo quickinstall cargo-nextest --version 0.9.28
+          cargo quickinstall cargo-llvm-cov --version 0.5.0
+
+      - name: Run tests
+        working-directory: ${{ matrix.directory }}
+        run: cargo +${{ matrix.toolchain }} make coverage --lcov --output-path lcov.info
```
Comment (kachkaev) [inline .github/workflows/rust.yml:224]: I wonder if it’s necessary to run tests again as a separate job. Not sure how easy it is in Rust, but in JS projects I’ve seen folks passing an env var to the matix. E.g.:

Node.js A: `yarn test`
Node.js B (LTS): `TEST_COVERAGE=true yarn test`
Node.js C: `yarn test`
Node.js D: `yarn test`

WDYT?

**Review by kachkaev — APPROVED** [review]
I left a couple of comments but I’m fine with trying things as is, just to start with something. If you want to go ahead with the merge, let’s create Asana tasks first so that we don’t forget things.

File: .github/workflows/rust.yml (hunk 1 above)
Comment (TimDiekmann) [reply .github/workflows/rust.yml:199]: I'd prefer doing this to be done in a follow up as this PR is unrelated

File: .github/workflows/rust.yml (hunk 2 above)
Comment (TimDiekmann) [reply .github/workflows/rust.yml:224]: I plan to merge coverage and test jobs as soon as we have carried forward flags so we don't need to run coverage unconditionally

**Pull request merged**